src/core/ext/transport/chaotic_good/frame.cc

```diff
@@ -86,32 +115,35 @@ class FrameSerializer {
   void MaybeCommitLast() {
     if (last_added_ == nullptr) return;
     *last_added_ = output_.Length() - length_at_last_added_;
-    if (output_.Length() % 64 != 0) {
-      output_.Append(kZeroSlice->RefSubSlice(0, 64 - output_.Length() % 64));
-    }
   }
 
   FrameHeader header_;
 
   uint32_t* last_added_ = nullptr;
   size_t length_at_last_added_;
   SliceBuffer output_;
+  BitSet<4> flags;  // Each bit represents a flag.
 };
 
 class FrameDeserializer {
  public:
   FrameDeserializer(const FrameHeader& header, SliceBuffer& input)
       : header_(header), input_(input) {}
   const FrameHeader& header() const { return header_; }
-  // If called, must be called before ReceiveMessage, ReceiveTrailers
+  // If called, must be called before ReceiveMessage, ReceivePadding,
+  // ReceiveTrailers, Finish.
   absl::StatusOr<SliceBuffer> ReceiveHeaders() {
     return Take(header_.header_length);
   }
-  // If called, must be called before ReceiveTrailers
+  // If called, must be called before ReceivePadding, ReceiveTrailers, Finish.
```

Comment:
Since messages are not part of the framing system anymore, I think we should remove ReceiveMessage (and therefore ReceivePadding) from this type.

Reply:
(instead we should add some helper functions to get the message lengths)